simples3.go

```diff
@@ -160,16 +163,20 @@ func (s3 *S3) getClient() *http.Client {
 }
 
 func (s3 *S3) getURL(bucket string, args ...string) (uri string) {
+	path := bucket
+	if len(args) > 0 {
+		path += "/" + strings.Join(args, "/")
+	}
+	// need to encode special characters in the path part of the URL
+	encodedPath := encodePath(path)
+
 	if len(s3.Endpoint) > 0 {
-		uri = s3.Endpoint + "/" + bucket
+		uri = s3.Endpoint + encodedPath
```

Comment:
The TestCustomEndpoint unit test seems to be failing, could you please check it?

```
--- FAIL: TestCustomEndpoint (0.00s)
    simples3_test.go:340: S3.SetEndpoint() got = https://example.com
    simples3_test.go:346: S3.SetEndpoint() got = http://localhost:9000
    simples3_test.go:352: S3.SetEndpoint() got = https://example.com
```